### ISSUE TITLE
Implement training loop for binary speaker verification

### ISSUE THREAD

**Issue by lelandsion** [lelandsion/music-singer-recognition#26]
Completion Criteria: Write the code to train the model to classify whether two audio samples belong to the same speaker, including forward pass, loss computation, and backpropagation.
